components/CategoryGrid.tsx

```diff
@@ -0,0 +1,79 @@
+import Image from "next/image";
+import { useState, useRef, useEffect } from "react";
+import { useInView } from 'react-intersection-observer';
+import Link from "next/link";
+import GridVideo from "./GridVideo";
+import useScreenWidth from '../hooks/useScreenWidth';
+
+const CategoryGrid = () => {
+    const [isHovering, setIsHovering] = useState(false);
+    const [cursorPosition, setCursorPosition] = useState({ x: 0, y: 0 });
+    const [scrollPosition, setScrollPosition] = useState(0);
+    const [sectionPosition, setSectionPosition] = useState({top: 0, left:0});
+    const sectionRef = useRef<HTMLInputElement>(null);
+    const [smallScreen, setSmallScreen] = useState(false);
+    const screenWidth = useScreenWidth();
+
+  const handleMouseEnter = () => {
+    setIsHovering(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovering(false);
+  };
+
+  const handleMouseMove = (e:any) => {
```

Comment:
For the event you can import the `MouseEvent` type from `react`

Check this article for reference: https://www.newline.co/@bespoyasov/how-to-handle-mouse-events-in-react-typescript-application--ba9bc920